options.go

```diff
@@ -0,0 +1,55 @@
+package cfdns
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/simplesurance/cfdns/logs"
+	"github.com/simplesurance/cfdns/logs/textlogger"
+	"golang.org/x/time/rate"
+)
+
+type Option func(*settings)
+
+type settings struct {
+	ratelim    *rate.Limiter
+	logger     *logs.Logger
+	httpClient *http.Client
+	logSuccess bool
+}
+
+func applyOptions(opts ...Option) *settings {
+	ret := settings{
+		ratelim:    rate.NewLimiter(rate.Every(defaultRequestInterval), 1),
+		logger:     logs.New(textlogger.New(os.Stdout, os.Stderr)),
+		httpClient: http.DefaultClient,
+	}
+	for _, opt := range opts {
+		opt(&ret)
+	}
+
+	return &ret
+}
+
+func WithRateLimiter(ratelim *rate.Limiter) Option {
+	return func(s *settings) {
+		s.ratelim = ratelim
+	}
+}
+
+func WithLogger(logger *logs.Logger) Option {
```

Comment:
The idea is that the users will provide a `Driver`, that specifies only where the log is sent to, so only the `Send` method has to be implemented, like in sisu: https://github.com/simplesurance/sisu/pull/22120/files#diff-fed0e6864e02dab8010f395c4e955bd919bf518997fceec8755ba71be43895be

The reason is that the code that receives the log request should imho be part of the log library, so it can collect information, and that can be not only what is passed as parameter, but also runtime data, like caller, maybe call stack or more. The driver receives an `Event` object with all data about that has to be logged.

Reply:
got it, sounds great